recipes/unicorn/all/conanfile.py

```diff
@@ -64,11 +59,21 @@ def configure(self):
 
     @property
     def _needs_jwasm(self):
-        return self.settings.os == "Windows" and self.settings.compiler != "Visual Studio"
+        return self.settings.os == "Windows" and not is_msvc(self)
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def build_requirements(self):
         if self._needs_jwasm:
-            self.build_requires("jwasm/2.13")
+            self.tool_requires("jwasm/2.13")
+
+    def package_id(self):
+        # normalize the supported_archs option (sorted+comma separated)
+        self.info.options.supported_archs = ",".join(self._supported_archs)
```

Comment:
I opened the issue https://github.com/conan-io/conan/issues/12700 which should try give better support on this.